extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/MutinyVertxClientFactory.java

```diff
@@ -0,0 +1,70 @@
+package io.quarkus.vault.runtime.client;
+
+import static io.quarkus.vault.runtime.client.CertificateHelper.createSslContext;
+import static io.quarkus.vault.runtime.client.CertificateHelper.createTrustManagers;
+import static io.quarkus.vault.runtime.config.VaultAuthenticationType.KUBERNETES;
+import static io.quarkus.vault.runtime.config.VaultRuntimeConfig.KUBERNETES_CACERT;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.vault.VaultException;
+import io.quarkus.vault.runtime.config.VaultRuntimeConfig;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.ext.web.client.WebClient;
+
+public class MutinyVertxClientFactory {
+
+    private static final Logger log = Logger.getLogger(MutinyVertxClientFactory.class.getName());
+
+    public static WebClient createHttpClient(VaultRuntimeConfig serverConfig) {
```

Comment:
Once done, the WebClient should be closed. Of course, if you want to keep it around until the application termination, no problem.

Reply:
this should be closed ideally from `VaultManager.reset()`
okhttp states that [shutdown isn't necessary](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#shutdown-isnt-necessary).
so when I switched to okhttp, I only nulled-out the vault manager singleton.
may be in certx we can do better?

Reply:
Yes, I would recommend closing the client. Closing releases resources and closes in-flight connections